packages/nuxt/test/runtime/utils/route-extraction.ts

```diff
@@ -0,0 +1,311 @@
+        modules: new Set(['app.vue', 'pages/user/profile.vue']),
+        currentUrl: '/user/profile',
+        buildTimePagesData: [
+          createMockRouteData({
+            name: 'user-profile',
+            path: '/user/profile',
+            file: '/app/pages/user/profile.vue',
+          }),
+        ],
+        expected: { parametrizedRoute: '/user/profile' },
+      },
+      {
+        description: 'dynamic route with brackets',
+        modules: new Set(['app.vue', 'pages/test-param/[param].vue']),
+        currentUrl: '/test-param/123',
+        buildTimePagesData: [
+          createMockRouteData({
+            name: 'test-param-param',
+            path: '/test-param/:param()',
+            file: '/app/pages/test-param/[param].vue',
+          }),
+        ],
+        expected: { parametrizedRoute: '/test-param/:param()' },
+      },
+      {
+        description: 'nested dynamic route',
+        modules: new Set(['app.vue', 'pages/test-param/user/[userId].vue']),
+        currentUrl: '/test-param/user/456',
+        buildTimePagesData: [
+          createMockRouteData({
+            name: 'test-param-user-userId',
+            path: '/test-param/user/:userId()',
+            file: '/app/pages/test-param/user/[userId].vue',
+          }),
+        ],
+        expected: { parametrizedRoute: '/test-param/user/:userId()' },
+      },
+    ])('should match $description', ({ modules, currentUrl, buildTimePagesData, expected }) => {
+      const result = extractParametrizedRouteFromContext(modules, currentUrl, buildTimePagesData);
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('different folder structures', () => {
+    it.each([
+      {
+        description: 'views folder instead of pages',
+        folderName: 'views',
+        modules: new Set(['app.vue', 'views/dashboard.vue']),
+        routeFile: '/app/views/dashboard.vue',
+        routePath: '/dashboard',
+      },
+      {
+        description: 'routes folder',
+        folderName: 'routes',
+        modules: new Set(['app.vue', 'routes/api/users.vue']),
+        routeFile: '/app/routes/api/users.vue',
+        routePath: '/api/users',
+      },
+      {
+        description: 'src/pages folder structure',
+        folderName: 'src/pages',
+        modules: new Set(['app.vue', 'src/pages/contact.vue']),
+        routeFile: '/app/src/pages/contact.vue',
+        routePath: '/contact',
+      },
+    ])('should work with $description', ({ modules, routeFile, routePath }) => {
+      const buildTimePagesData = [
+        createMockRouteData({
+          name: 'test-route',
+          path: routePath,
+          file: routeFile,
+        }),
+      ];
+
+      const result = extractParametrizedRouteFromContext(modules, routePath, buildTimePagesData);
+      expect(result).toEqual({ parametrizedRoute: routePath });
+    });
+  });
+
+  describe('multiple routes matching', () => {
+    it('should find the correct route when multiple routes exist', () => {
```

Comment:
sticking with this example: If I have two routes,

```
file: '/app/pages/test-param/user/[userId].vue',
file: '/app/pages/test-param/user/abc.vue',
```

do we still correctly identify the respective routes for urls `/test-param/user/123` and `/test-param/user/abc`?

(Sorry in case I missed such an example in already existing tests)

Reply:
Related but not also rather a question: Does this handle [catch-all routes](https://nuxt.com/docs/guide/directory-structure/pages#catch-all-route)?